src/production/__tests__/CodeSplitting.test.ts

```diff
@@ -0,0 +1,444 @@
+import { IProductionContext } from '../ProductionContext';
+import { CodeSplittingPhase } from '../phases/CodeSplittingPhase';
+import { WarmupPhase } from '../phases/WarmupPhase';
+import { ProdPhasesTestEnv, prodPhasesEnv } from './testUtils';
+
+describe('Code Splitting test', () => {
+  let env: ProdPhasesTestEnv;
+  const test = async (files: Record<string, string>): Promise<IProductionContext> => {
+    env = prodPhasesEnv(
+      {
+        entry: 'index.ts',
+      },
+      files,
+    );
+    return await env.run([WarmupPhase, CodeSplittingPhase]);
+  };
+  afterEach(() => {
+    if (env) env.destroy();
+    env = undefined;
+  });
+
+  it('should have splitEntry', async () => {
+    const context = await test({
+      'index.ts': `
+        import './foo';
+        async function load() {
+          const dynamicFunc = await import('./bar');
+        }
+      `,
+      'foo.ts': `
+        export default function() {}
+      `,
+      'bar.ts': `
+        export default function() {}
+      `,
+    });
+    const { modules, splitEntries: { entries } } = context;
+    expect(modules).toHaveLength(3);
+    expect(entries).toHaveLength(1);
+    expect(entries[0].modules).toHaveLength(1);
+    expect(entries[0].modules[0].getShortPath()).toEqual('default/bar.js');
+    expect(entries[0].references).toHaveLength(1);
+    expect(entries[0].references[0].module.getShortPath()).toEqual('default/index.js');
+  });
+
+  it('should have splitEntry with submodules', async () => {
+    const context = await test({
+      'index.ts': `
+        async function load() {
+          const dynamicFunc = await import('./bar');
+        }
+      `,
+      'bar.ts': `
+        import './foo';
+        export default function() {}
+      `,
+      'foo.ts': `
+        export default function() {}
+      `,
+    });
+    const { modules, splitEntries: { entries } } = context;
+    expect(modules).toHaveLength(3);
+    expect(entries).toHaveLength(1);
+    expect(entries[0].modules).toHaveLength(2);
+    expect(entries[0].modules[0].getShortPath()).toEqual('default/bar.js');
+    expect(entries[0].modules[1].getShortPath()).toEqual('default/foo.js');
+    expect(entries[0].references).toHaveLength(1);
+    expect(entries[0].references[0].module.getShortPath()).toEqual('default/index.js');
+  });
+
+  it('should drop imported module thats required directly somewhere else', async () => {
+    const context = await test({
+      'index.ts': `
+        import './foo';
+        import { test } from './test';
+
+        async function load() {
+          const dynamicFunc = await import('./bar');
+        }
+
+        console.log(test);
+      `,
+      'foo.ts': `
+        export function foo() {}
+      `,
+      'bar.ts': `
+        import { test } from './test';
```

Comment:
This module isn't in use

Reply:
yes it is, in index.ts => `console.log(test)`

Reply:
true, extended the test!